src/viztracer/modules/vcompressor/vc_dump.c

```diff
@@ -146,6 +147,57 @@ int dump_parsed_trace_events(PyObject* trace_events, FILE* fptr)
         }
     }
 
+    // Iterate through instant events
+    ppos = 0;
+    PyObject* dumps_func = NULL;
```

Comment:
Could you move the declaration on the top of the function? We should only declare variables at the top of a code block.